flow360/component/simulation/translator/solver_translator.py

```diff
@@ -0,0 +1,180 @@
+)
+from flow360.component.simulation.simulation_params import SimulationParams
+from flow360.component.simulation.translator.utils import (
+    get_attribute_from_first_instance,
+    preprocess_input,
+)
+from flow360.component.simulation.unit_system import LengthType
+
+
+def to_dict(input_params):
+    return input_params.model_dump(by_alias=True, exclude_none=True)
+
+
+def to_dict_without_unit(input_params):
+    unit_keys = {"value", "units"}
+    output_dict = to_dict(input_params)
+    for key, value in output_dict.items():
+        output_dict[key] = (
+            value["value"] if isinstance(value, dict) and value.keys() == unit_keys else value
+        )
+    return output_dict
+
+
+def remove_empty_keys(input_dict):
+    # TODO: implement
+    return input_dict
+
+
+def init_output_attr_dict(obj_list, class_type):
+    return {
+        "animationFrequency": get_attribute_from_first_instance(obj_list, class_type, "frequency"),
+        "animationFrequencyOffset": get_attribute_from_first_instance(
+            obj_list, class_type, "frequency_offset"
+        ),
+        "outputFormat": get_attribute_from_first_instance(obj_list, class_type, "output_format"),
+    }
+
+
+def get_output_fields(obj_list, class_type):
+    return get_attribute_from_first_instance(obj_list, class_type, "output_fields").model_dump()[
+        "items"
+    ]
+
+
+def merge_unique_item_lists(list1: list, list2: list) -> list:
+    # TODO: implement
+    return list1 + list2
+
+
+@preprocess_input
+def get_solver_json(
+    input_params: Union[str, dict, SimulationParams],
+    mesh_unit: LengthType.Positive,
+):
+    """
+    Get the surface mesh json from the simulation parameters.
+    """
+    outputs = input_params.outputs
+    translated = {
+        "boundaries": {},
+        "volumeOutput": init_output_attr_dict(outputs, VolumeOutput),
+        "surfaceOutput": init_output_attr_dict(outputs, SurfaceOutput),
+        "sliceOutput": init_output_attr_dict(outputs, SliceOutput),
+    }
+    translated["volumeOutput"].update(
+        {"outputFields": get_output_fields(outputs, VolumeOutput), "computeTimeAverages": False}
+    )
+    translated["surfaceOutput"].update(
+        {
+            "writeSingleFile": get_attribute_from_first_instance(
+                outputs, SurfaceOutput, "write_single_file"
+            ),
+            "surfaces": {},
+            "computeTimeAverages": False,
+        }
+    )
+    translated["sliceOutput"].update({"slices": {}})
+
+    bd = translated["boundaries"]
+
+    op = input_params.operating_condition
+    translated["freestream"] = {
+        "alphaAngle": op.alpha.to("degree").v.item(),
+        "betaAngle": op.beta.to("degree").v.item(),
+        "Mach": op.velocity_magnitude.v.item(),
+        "Temperature": op.thermal_state.temperature.to("K").v.item(),
+        "muRef": op.thermal_state.mu_ref(mesh_unit),
+    }
+    if "reference_velocity_magnitude" in op and op.reference_velocity_magnitude:
```

Comment:
This `"reference_velocity_magnitude" in op` works? op is an object and "reference_velocity_magnitude" is string?

Reply:
Right. I think I meant to write __fields__